docs/design/generics/details.md

```diff
@@ -3688,14 +3132,21 @@ fn SortContainer
 
 In contrast to a [rewrite constraint](#rewrite-constraints) or a
 [same-type constraint](#same-type-constraints), this does not say what type
-`ElementType` exactly is, just that it must satisfy requirements of some facet
-type.
-
-> **Note:** `Container` defines `ElementType` as having type `type`, but
-> `ContainerType.ElementType` has type `Ordered`. This is because
-> `ContainerType` has type `Container where .ElementType impls Ordered`, not
-> `Container`. This means we need to be a bit careful when talking about the
-> type of `ContainerType` when there is a `where` clause modifying it.
+`ElementType` exactly is, just that it must satisfy the requirements of some
+facet type.
+
+The specific case of a clause of the form
+`where .AssociatedFacet impls AConstraint`, where the constraint is applied to a
+direct associated facet member of the facet type being constrained (similar to
+the restriction on [rewrite constraints](#rewrite-constraints)), gets special
+treatment. In this case, the type of the associated facet is
+[combined](#combining-interfaces-by-anding-facet-types) with the constraint. In
+the above example, `Container` defines `ElementType` as having type `type`, but
+`ContainerType.ElementType` has type `type & Ordered` (which is equivalent to
+`Ordered`). This is because `ContainerType` has type
+`Container where .ElementType impls Ordered`, not `Container`. This means we
+need to be a bit careful when talking about the type of `ContainerType` when
+there is a `where` clause modifying it.
```

Comment:
Maybe we should discuss further when @chandlerc gets back from vacation. It seems a little inconsistent to me to say that we need different syntax (`=` vs `==`) for type-changing equality constraint, but for `impls` we will just look at whether the left-hand side is written as a designator to determine whether it affects the type of the member. I think it's fine for now, though, and the new text seems like a more precise form of the old note, so I don't think we should block this PR on it.

Reply:
Added "future work"